packages/@tinacms/toolkit/src/components/ActiveFieldIndicator.tsx

```diff
@@ -0,0 +1,184 @@
+/**
+
+Copyright 2021 Forestry.io Holdings, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import * as React from 'react'
+import styled from 'styled-components'
+import { ChevronUpIcon, ChevronDownIcon } from '../packages/icons'
+import { useEvent } from '../packages/react-core/use-cms-event'
+import {
+  FieldHoverEvent,
+  FieldFocusEvent,
+} from '../packages/fields/field-events'
+import { useFieldReference } from '../hooks/use-field-reference'
+
+const IndicatorWrap = styled.div`
+  position: fixed;
+  z-index: var(--tina-z-index-3);
+  left: 0;
+  padding: 0.5rem 0;
+  margin-left: var(--tina-sidebar-width);
+  width: calc(100% - var(--tina-sidebar-width));
+  text-align: center;
+  top: ${(props: { position?: 'top' | 'bottom' }) =>
+    props.position === 'top' ? 0 : 'auto'};
+  bottom: ${(props: { position?: 'top' | 'bottom' }) =>
+    props.position === 'top' ? 'auto' : 0};
+`
+
+const ArrowWrap = styled.div`
+  display: inline-block;
+  fill: white;
+  background-color: var(--tina-color-primary);
+  border-radius: 50%;
+  box-shadow: 0 0 10px -5px;
+`
+
+const AboveViewportIndicator = () => {
+  return (
+    <IndicatorWrap position="top">
+      <ArrowWrap>
+        <ChevronUpIcon />
+      </ArrowWrap>
+    </IndicatorWrap>
+  )
+}
+
+const BelowViewportIndicator = () => {
+  return (
+    <IndicatorWrap position="bottom">
+      <ArrowWrap>
+        <ChevronDownIcon />
+      </ArrowWrap>
+    </IndicatorWrap>
+  )
+}
+
+const useScrollToFocusedField = () => {
```

Comment:
[scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) is a pretty slick API which may take care of all of this automatically. I remember there may be some footguns when there are nested scrollable elements, but I'm not sure either approach will handle those well.

Reply:
I'm pretty happy w/ the scroll behavior. If `scrollIntoView` weren't experimental it might be worth looking into for the sake of simplifying this code, but as it is the current implementation is probably more future-proof.